solr/solrj/src/java/org/apache/solr/client/solrj/impl/ClusterStateProvider.java

```diff
@@ -108,8 +112,7 @@ default DocCollection getCollection(String name) throws IOException {
   /** Obtain a cluster property, or the default value if it doesn't exist. */
   default <T> T getClusterProperty(String key, T defaultValue) {
     @SuppressWarnings({"unchecked"})
-    T value = (T) getClusterProperties().get(key);
-    if (value == null) return defaultValue;
+    T value = (T) getClusterProperties().getOrDefault(key, defaultValue);
     return value;
```

Comment:
one line; no need to declare

Reply:
(waiting)

Reply:
I am unable to remove the declare else the build fails with 

Task :solr:solrj:compileJava
/Users/aparna.suresh/blt/upstream/configurable-clusterstatus/solr/solrj/src/java/org/apache/solr/client/solrj/impl/ClusterStateProvider.java:114: warning: [unchecked] unchecked cast
  return (T) getClusterProperties().getOrDefault(key, defaultValue);
                         ^
 required: T
 found:  Object
 where T is a type-variable:
  T extends Object declared in method <T>getClusterProperty(String,T)
error: warnings found and -Werror specified
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error
1 warning

Reply:
Wow, you're right; even the code for getOrDefault suggests it's necessary.  Okay.